Compiler/script2/cs_scanner.cpp

```diff
@@ -544,14 +545,34 @@ AGS::ErrorType AGS::Scanner::ReadInStringLit(std::string &symstring, std::string
             ErrorType retval = SkipWhitespace();
             if (retval < 0) return retval;
 
-            if ('"' != Peek())
+
+            if ('"' != Peek() ||
+                kNewSectionLitPrefix == valstring.substr(0, kNewSectionLitPrefix.length()))
             {
                 _inputStream.seekg(pos_before_skip);
                 _lineno = lineno_before_skip;
-                return kERR_None; // End of string
+                return kERR_None;
             }
 
             // Another string literal follows
+            // Tentatively read ahead to check whether it's a new section marker
+            pos_before_skip = _inputStream.tellg();
+            lineno_before_skip = _lineno;
+            Get(); // Eat leading '"'
+            char *tbuffer = new char[kNewSectionLitPrefix.length() + 1];
```

Comment:
I think you may avoid repeatedly allocating extra buffer here, since kNewSectionLitPrefix is constant, having a large enough buffer allocated on stack could be enough.

Reply:
It's not a code I'm the most proud of. My trouble was with the `std::istring::get()` call: it takes a `char *` in the first argument, and I didn't find a variant that takes a `std::string` instead. Bummer. So at first I tried to force a string into the function call:
```
std::string buffer;
buffer.reserve(…); // so that its internal buffer is long enough
… get(&(buffer[0]), …) …
```
but after that sequence, the content __looked__ okay but `buffer.length()` reported the wrong length. Well okay, it's probably a dirty coding to begin with, so let's behave and not do that, Peter.  🙄

So that's when I allocated them their d*mn `char[]`, moved the result into a std::string and freed the `char[]` directly afterwards. 

Okay, after a night's sleep, I'll try to get that more elegant.
Okay, I'll come up with something more elegant.